packages/cli/generation/ir-generator/src/converters/services/convertHttpResponse.ts

```diff
@@ -53,3 +75,95 @@ function constructStreamingResponseChunkType(
         return StreamingResponseChunkType.json(file.parseTypeReference(typeReference));
     }
 }
+
+function typeReferenceHasProperty(
+    typeReference: TypeReference,
+    property: string,
+    file: FernFileContext,
+    typeResolver: TypeResolver
+): boolean {
+    if (typeReference.type === "container" && typeReference.container.type === "optional") {
+        return typeReferenceHasProperty(typeReference.container.optional, property, file, typeResolver);
+    }
+    if (typeReference.type === "named") {
+        const resolvedType = typeResolver.resolveNamedTypeOrThrow({
+            referenceToNamedType: typeReference.name.originalName,
+            file,
+        });
+        switch (resolvedType._type) {
+            case "container":
+                return typeReferenceHasProperty(resolvedType.originalTypeReference, property, file, typeResolver);
+            case "named":
+                if (isRawObjectDefinition(resolvedType.declaration)) {
+                    return rawObjectSchemaHasProperty(resolvedType.declaration, property, file, typeResolver);
+                }
+                throw new Error("A custom response property is only supported for objects");
+        }
+    }
```

Comment:
do we need to handle aliases here?

Reply:
Aliases are already handled by the `TypeResolver` we're using - it only ever returns the underlying type. You can see that handled [here](https://github.com/fern-api/fern/blob/ff01eefce12d16d0685bc3146cd7fa84038dc82e/packages/cli/generation/ir-generator/src/resolvers/TypeResolver.ts#L256).

Reply:
thanks!